openfe/protocols/openmm_afe/equil_solvation_afe_method.py

```diff
@@ -0,0 +1,575 @@
+                      f"repeat {i} generation 0"),
+            )
+            for i in range(self.settings.alchemsampler_settings.n_repeats)
+        ]
+
+        vacuum_units = [
+            AbsoluteVacuumTransformUnit(
+                # These don't really reflect the actual transform
+                # Should these be overriden to be ChemicalSystem{smc} -> ChemicalSystem{} ?
+                stateA=stateA, stateB=stateB,
+                settings=self.settings,
+                alchemical_components=alchem_comps,
+                generation=0, repeat_id=i,
+                name=(f"Absolute Solvation, {alchname} vacuum leg: "
+                      f"repeat {i} generation 0"),
+            )
+            for i in range(self.settings.alchemsampler_settings.n_repeats)
+        ]
+
+        return solvent_units + vacuum_units
+
+    def _gather(
+        self, protocol_dag_results: Iterable[gufe.ProtocolDAGResult]
+    ) -> Dict[str, Dict[str, Any]]:
+        # result units will have a repeat_id and generation
+        # first group according to repeat_id
+        unsorted_solvent_repeats = defaultdict(list)
+        unsorted_vacuum_repeats = defaultdict(list)
+        for d in protocol_dag_results:
+            pu: gufe.ProtocolUnitResult
+            for pu in d.protocol_unit_results:
+                if not pu.ok():
+                    continue
+                if pu.outputs['simtype'] == 'solvent':
+                    unsorted_solvent_repeats[pu.outputs['repeat_id']].append(pu)
+                else:
+                    unsorted_vacuum_repeats[pu.outputs['repeat_id']].append(pu)
+
+        repeats: dict[str, dict[str, list[gufe.ProtocolUnitResult]]] = {
+            'solvent': {}, 'vacuum': {},
+        }
+        for k, v in unsorted_solvent_repeats.items():
+            repeats['solvent'][str(k)] = sorted(v, key=lambda x: x.outputs['generation'])
+
+        for k, v in unsorted_vacuum_repeats.items():
+            repeats['vacuum'][str(k)] = sorted(v, key=lambda x: x.outputs['generation'])
+        return repeats
+
+
+class AbsoluteVacuumTransformUnit(BaseAbsoluteTransformUnit):
+    def _get_components(self):
+        """
+        Get the relevant components for a vacuum transformation.
+
+        Returns
+        -------
+        alchem_comps : list[Component]
+          A list of alchemical components
+        solv_comp : None
+          For the gas phase transformation, None will always be returned
+          for the solvent component of the chemical system.
+        prot_comp : Optional[ProteinComponent]
+          The protein component of the system, if it exists.
+        small_mols : list[SmallMoleculeComponent]
+          A list of SmallMoleculeComponents to add to the system.
+        """
+        stateA = self._inputs['stateA']
+        alchem_comps = self._inputs['alchemical_components']
+
+        _, prot_comp, small_mols = system_validation.get_components(stateA)
```

Comment:
This won't work if you tried to shoehorn in spectator molecules - the SMC here should always just be the alchemical component since it's a correction.